src/components/VideoPlayer/BaseVideoPlayer.tsx

```diff
@@ -94,8 +95,14 @@ function BaseVideoPlayer({
     }, [isCurrentlyURLSet, isPlaying, pauseVideo, playVideo, updateCurrentlyPlayingURL, url, videoResumeTryNumber]);
 
     const showPopoverMenu = (event?: GestureResponderEvent | KeyboardEvent) => {
-        setIsPopoverVisible(true);
         videoPopoverMenuPlayerRef.current = videoPlayerRef.current;
+        videoPlayerRef.current?.getStatusAsync().then((status) => {
+            if (!('rate' in status && status.rate)) {
+                return;
+            }
+            setIsPopoverVisible(true);
+            setCurrentPlaybackSpeed(status.rate as PlaybackSpeed);
+        });
```

Comment:
@hungvu193 

> Please see the video, I don't think it's working for IOS.

This bug maybe related with delay of this function.  Run time of this function is affected by some factors like video size, quality and network status.

Could you please check changes?

> I got an infinite loop render also

I can't sure about the infinite loop render you mentioned. Could you please explain in more detail?

Reply:
I got the loop few times so I can not give you a consistent reproducible steps but let me try.